### PR TITLE
New image uploader

### PR DESCRIPTION
**Pull request by minhngocd**
This new image uploader is used for featured images (including Lead Images, Person Images, Featured Images) and hence do not allow SVG file uploads.

Most non-inline images currently use ImageUploader, but only ever send a specific size over to publishing API, resulting in an error or some other unexpected behaviour when SVGs are uploaded.
The logic required to add SVG blocking using in conjunction with the other ImageUploader and also managing error messages seemed overly complex as compared to adding another uploader for these types of Featured Images which encompasses the resizing as well as the extension blocking and its validation messages.

SVGs should be used for graphs and charts and not as person images, featured images, and lead images, so we have introduced a new uploader to support this type of images.

https://trello.com/c/ZjWXXW93/1438-add-wrong-file-type-errors-to-topical-events-logo-uploads

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

Follow [these steps](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html) if you are doing a Rails upgrade.

### CODE REVIEW

**Review by ryanb-gds — APPROVED** [review]
Looks good to me

File: app/uploaders/featured_image_uploader.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+class FeaturedImageUploader < WhitehallUploader
```
Comment (ollietreend) [inline app/uploaders/featured_image_uploader.rb:1]: Generally speaking, I feel like the words "feature" and "featured" are a bit over-used in Whitehall. And it's making me wonder if there could be a slightly more intention revealing name for this uploader class.

For somebody new coming into this codebase:

- The `FeaturedImageUploader` is used for `Feature#image` – and with a name this similar, it sort of sounds like it was made specifically for this use case. _Is it supposed to be coupled to this model?_ 🤔
- But it's also used on `PromotionalFeatureItem#image`. _So maybe it's supposed to be used for images on "featured" things?_ 🤔
- But then it's also used on `Person#image`. _Does that mean it's been re-purposed for something it wasn't originally intended?_ 🤔

I suspect you've probably already thought through different naming options before landing on this. So I'm totally happy for you to stick with this name. This is not a strongly held opinion 🙂 but the only suggestion I might throw into the ring is to maybe call it `BitmapImageUploader` – which would hopefully reveal the intention of this being used for uploading images that cannot be vectors.

**Review by ollietreend — COMMENTED** [review]
Lovely work @minhngocd and @jyoung-gds 😎

Just a small opinion about naming.

File: app/uploaders/featured_image_uploader.rb (hunk 1 above)
Comment (minhngocd) [reply app/uploaders/featured_image_uploader.rb:1]: We struggled a lot with this as well and was also teetering between BitmapImageUploader and FeaturedImageUploader. We thought about if people were implementing new document types with images, how do they know when to use ImageUploader and BitmapImageUploader - so we thought that having FeaturedImageUploader would be clearer about when it should be used (and maybe renaming ImageUploader to InlineImageUploader or something equivalent). 

But agree it's not the best name - we have a thread trying to ask design folks what they would call these types of images as opposed to the inline ones and "Featured" was one of the things that was suggested.

Comment (minhngocd) [reply app/uploaders/featured_image_uploader.rb:1]: I'm either way writing up a tech debt card in the backlog to maybe rethink image scaling for these types of images (backend vs frontend) - maybe i'll also include in there how to categorise and name image uploaders?

Comment (ollietreend) [reply app/uploaders/featured_image_uploader.rb:1]: Thanks for explaining @minhngocd – that makes sense! ✨ ✅

**Review by ollietreend — APPROVED** [review]
Looks good to me 👍🏻

**Pull request merged**